### PR TITLE
Use uv in CI

### PR DESCRIPTION
**Pull request by dwreeves**
## Description

The CI takes a really long time to run sometimes:

image

I'm hoping that switching to `uv` will make it run significantly faster.

## Related Issue(s)

n/a

## Breaking Change?

nope

## Checklist

n/a

### CODE REVIEW

**Comment by dwreeves** [comment]
A note: The workflows running for this PR are the old one, so if this passes it doesn't mean the actual changes to the Github Action workflow succeeded.

**Comment by dwreeves** [comment]
Seems to run a a little bit faster (shaves about 30 seconds off each run), but Hatch (not uv) is still getting stuck on resolving the Airflow 2.7 dependencies.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -2,9 +2,9 @@ name: test
 
 on:
   push: # Run on pushes to the default branch
-    branches: [main]
+    branches: [main, switch-ci-to-uv]
```
Comment (tatiana) [inline .github/workflows/test.yml:5]:
```suggestion
    branches: [main]
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -2,9 +2,9 @@ name: test
...
   pull_request_target: # Also run on pull requests originated from forks
-    branches: [main]
+    branches: [main, switch-ci-to-uv]
```
Comment (tatiana) [inline .github/workflows/test.yml:7]:
```suggestion
    branches: [main]
```

**Review by tatiana — APPROVED** [review]
Every second counts, thanks a lot, @dwreeves for checking the impact of these changes.

I've reverted the branch reference and we're good to merge this 🎉

**Pull request merged**